docs/contract_types.rst

```diff
@@ -2,6 +2,23 @@
 Contract types
 ==============
 
+Shared options
+--------------
+
+The following options can be used in all contracts.
+
+- ``ignore_imports``: Optional list of imports, each in the form ``mypackage.foo.importer -> mypackage.bar.imported``.
```

Comment:
(Not related to documentation, but had to write this somewhere!) It would be great to have the following test coverage:

- Unit tests for wildcarded ignore_imports for each contract type (see tests/unit/contracts).
- Functional test coverage in tests/functional/test_lint_imports.py. I suggest just adding an illegal import, together with a wildcarded `ignore_imports` that covers it, to the contract in `setup.cfg` in that folder (which is used for checking contract adherence).

Reply:
@seddonym I can work further on the pure functions but I guess I don't have the time for extended tests. I really see the point of them, but it consumes a lot of time. Would you maybe just add them?